packages/shared/lib/services/connection.service.ts

```diff
@@ -785,56 +785,57 @@ class ConnectionService {
         limit?: number;
         page?: number | undefined;
     }): Promise<{ connection: DBConnection; end_user: DBEndUser | null; active_logs: [{ type: string; log_id: string }]; provider: string }[]> {
-        const subQuery = db.knex
+        const query = db.knex
             .from<Connection>(`_nango_connections`)
-            .select(
+            .select<{ connection: DBConnection; end_user: DBEndUser | null; active_logs: [{ type: string; log_id: string }]; provider: string }[]>(
                 db.knex.raw('row_to_json(_nango_connections.*) as connection'),
                 db.knex.raw('row_to_json(end_users.*) as end_user'),
                 db.knex.raw(`
-                  (SELECT COALESCE(json_agg(json_build_object(
-                      'type', type,
-                      'log_id', log_id
-                    )), '[]'::json)
-                    FROM ${ACTIVE_LOG_TABLE}
-                    WHERE _nango_connections.id = ${ACTIVE_LOG_TABLE}.connection_id
-                      AND ${ACTIVE_LOG_TABLE}.active = true
-                  ) as active_logs
+                    COALESCE(
+                        json_agg(
+                            json_build_object(
+                                'type', _nango_active_logs.type,
+                                'log_id', _nango_active_logs.log_id
+                            )
+                        ) FILTER (WHERE _nango_active_logs.id IS NOT NULL)
+                        , '[]'::json
+                    ) as active_logs
                `),
+                db.knex.raw('count(_nango_active_logs.id) as active_logs_count'),
                 '_nango_configs.provider'
             )
             .join('_nango_configs', '_nango_connections.config_id', '_nango_configs.id')
             .leftJoin('end_users', 'end_users.id', '_nango_connections.end_user_id')
+            .leftJoin(ACTIVE_LOG_TABLE, function () {
+                this.on(`${ACTIVE_LOG_TABLE}.connection_id`, '_nango_connections.id').andOn(`${ACTIVE_LOG_TABLE}.active`, db.knex.raw(true));
+            })
             .where({
                 '_nango_connections.environment_id': environmentId,
                 '_nango_connections.deleted': false
             })
-            .orderBy('_nango_connections.created_at', 'desc');
+            .orderBy('_nango_connections.created_at', 'desc')
+            .groupBy('_nango_connections.id', 'end_users.id', '_nango_configs.provider')
```

Comment:
TIL you could groupBy `.id` with a `row_to_json`, that was the main reason for the subQuery, thanks 🙏🏻